bluesky_kafka/tests/conftest.py

```diff
@@ -23,3 +39,109 @@ def bootstrap_servers(request):
 @pytest.fixture(scope="function")
 def hw(request):
     return ophyd.sim.hw()
+
+
+@pytest.fixture(scope="function")
+def msgpack_serializer(request):
+    return partial(msgpack.dumps, default=mpn.encode)
+
+
+@pytest.fixture(scope="function")
+def msgpack_deserializer(request):
+    return partial(msgpack.loads, object_hook=mpn.decode)
+
+
+@pytest.fixture(scope="function")
+def publisher(request, bootstrap_servers, msgpack_serializer):
+    return Publisher(
+        topic=TEST_TOPIC,
+        bootstrap_servers=bootstrap_servers,
+        key="kafka-unit-test-key",
+        # work with a single broker
+        producer_config={
+            "acks": 1,
+            "enable.idempotence": False,
+            "request.timeout.ms": 5000,
+        },
+        serializer=msgpack_serializer,
+        flush_on_stop_doc=True,
+    )
+
+
+@pytest.fixture(scope="function")
+def publisher2(request, bootstrap_servers, msgpack_serializer):
+    return Publisher(
+        topic=TEST_TOPIC2,
+        bootstrap_servers=bootstrap_servers,
+        key="kafka-unit-test-key",
+        # work with a single broker
+        producer_config={
+            "acks": 1,
+            "enable.idempotence": False,
+            "request.timeout.ms": 5000,
+        },
+        serializer=msgpack_serializer,
+        flush_on_stop_doc=True,
+    )
+
+
+@pytest.fixture(scope="function")
+def mongo_client(request):
+    mongobox = pytest.importorskip('mongobox')
+    box = mongobox.MongoBox()
+    box.start()
+    return box.client()
+
+
+@pytest.fixture(scope="function")
+def mongo_uri(request, mongo_client):
+    return f"mongodb://{mongo_client.address[0]}:{mongo_client.address[1]}"
+
+
+@pytest.fixture(scope="function")
+def md(request):
+    return {"numpy_data": {"nested": np.array([1, 2, 3])},
+            "numpy_scalar": np.float64(3),
+            "numpy_array": np.ones((3, 3))}
+
+
+@pytest.fixture(scope="function")
+def broker(request, mongo_uri):
```

Comment:
Could this fixture be called "data_broker" to differentiate from message brokers?

Reply:
Updated.